plugins/woocommerce/includes/class-wc-template-loader.php

```diff
@@ -529,12 +529,12 @@ private static function get_current_shop_view_args() {
 	 * For non-WC themes, this will setup the main shop page to be shortcode based to improve default appearance.
 	 *
 	 * @since 3.3.0
-	 * @param string $title Existing title.
-	 * @param int    $id ID of the post being filtered.
+	 * @param string   $title Existing title.
+	 * @param int|null $id ID of the post being filtered.
 	 * @return string
 	 */
-	public static function unsupported_theme_title_filter( $title, $id ) {
-		if ( self::$theme_support || ! $id !== self::$shop_page_id ) {
+	public static function unsupported_theme_title_filter( $title, $id = null ) {
```

Comment:
Suggested change to account for [a PHP 8.4 deprecation](https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated):

```suggestion
	public static function unsupported_theme_title_filter( $title, ?$id = null ) {
```